test/test_array_ops.cpp

```diff
@@ -51,9 +51,71 @@ namespace testing {
 // Neither TestCaseName nor TestName should contain underscore
 // https://github.com/google/googletest/blob/master/googletest/docs/primer.md
 // Use only Tensors and ops::Const() to provide input to the test op
+// Please ensure the alphabetical order while adding the test functions
 
-// PreventGradient op: an identity op that triggers an error if gradient is
-// requested
+// Test op: Fill
+TEST(ArrayOps, Fill) {
+  std::vector<std::vector<int>> input_sizes;  // 1-D or higher
+
+  input_sizes.push_back({2, 3, 4});
+  input_sizes.push_back({10, 10, 10});
+  input_sizes.push_back({1, 5});
+  input_sizes.push_back({0});
+  input_sizes.push_back({2, 5, 1, 3, 1});
+
+  vector<int> static_input_indexes = {0};  // has static input
+
+  for (auto const& input_size : input_sizes) {
+    Scope root = Scope::NewRootScope();
+
+    int input_dim = input_size.size();
+    Tensor shape(DT_INT32, TensorShape({input_dim}));
+    AssignInputValuesFromVector<int>(shape, input_size);
+
+    // 0-D(scalar) value to fill the returned tensor
+    Tensor input_data(DT_FLOAT, TensorShape({}));
+    AssignInputValuesRandom(input_data);
+
+    // Fill creates a tensor filled with scalar value
+    // 1-D shape of the output tensor
+    auto R = ops::Fill(root, shape, input_data);
+    vector<DataType> output_datatypes = {DT_FLOAT};
+    std::vector<Output> sess_run_fetchoutputs = {R};
+
+    OpExecuter opexecuter(root, "Fill", static_input_indexes, output_datatypes,
+                          sess_run_fetchoutputs);
+    opexecuter.RunTest();
+  }  // end of for loop
+}  // end of op Fill
+
+// Test op: ExpandDims, inserts a dimension of 1 into a tensor's shape
+TEST(ArrayOps, ExpandDims) {
+  Scope root = Scope::NewRootScope();
+
+  int dim1 = 2;
+  int dim2 = 3;
+
+  Tensor A(DT_FLOAT, TensorShape({dim1, dim2}));
+  AssignInputValuesRandom(A);
+
+  // axis at which the dimension will be inserted
+  // should be -rank-1 <= axis <= rank
+  auto axis = 0;
+
```

Comment:
I tested with axis = -1 and it passed. Is there a need for a separate test?

Reply:
I meant in a loop. you can have the same 'A' tensor of size 2x3, just varying the 'axis' variable is enough.

we should add it, to protect against regressions. even if it passes now, it could fail in the future.

Reply:
Done